### PR TITLE
38 39 pong improvement

### PR DESCRIPTION
**Pull request by C-Begley**
Angles are fixed and score is tracked

### CODE REVIEW

File: SmartBadge/pong.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ def __init__(self):
         self.direction = [1, 1]
 
     def reset(self):
+        self.x = 16
+        self.y = 16
```
Comment (miklasr) [inline SmartBadge/pong.py]: Why 16?
Could it be made a global constant?

File: SmartBadge/pong.py [hunk 2]
```diff
@@ -89,67 +91,83 @@ def __init__(self, disp, buttons, tim):
...
-                elif self.ball.y > self.player_2.y + self.player_2.height//3:
-                    self.ball.direction[1] = 0
+        if not self.game_over:
+
+            Right = self.game.collision_edge(self.ball, 0, 1)
+            Left = self.game.collision_edge(self.ball, 0, -1)
+            Up = self.game.collision_edge(self.ball, 1, 1)
+            Down = self.game.collision_edge(self.ball, 1, -1)
+        
+            if self.ball.x <= - self.ball.width or  self.ball.x >= 32 + self.ball.width:
```
Comment (miklasr) [inline SmartBadge/pong.py]: why 32? (same as above)

File: SmartBadge/pong.py [hunk 3]
```diff
@@ -89,67 +91,83 @@ def __init__(self, disp, buttons, tim):
...
-                self.ball.direction[0] = 1
-
-                if self.ball.y > self.player_1.y + 2*self.player_1.height//3:
-                    self.ball.direction[1] = 1
-                elif self.ball.y > self.player_1.y + self.player_1.height//3:
-                    self.ball.direction[1] = 0
+                    self.scores.update_text("{s1}-{s2}".format(s1 = self.score1, s2 = self.score2))
+                    self.ball.reset()
+            else:
+                if (self.ball.x + self.ball.width == 29 or self.ball.x - self.ball.width == 3):
```
Comment (miklasr) [inline SmartBadge/pong.py]: Hardcoded 29

File: SmartBadge/pong.py (hunk 1 above)
Comment (C-Begley) [reply SmartBadge/pong.py]: As it stands, these values are all hard coded as we have not fully decided the aspect ratio for the grid

Comment (miklasr) [reply SmartBadge/pong.py]: Could we at least have them as global constants, so they can be easily changed? The 16 on its own isn't very descriptive.

Comment (C-Begley) [reply SmartBadge/pong.py]: So, I'd like to tidy this up across all the files and I think this should be done after a merge to master

**Pull request merged**